s12/protobuf/proto/validator_helpers.go

```diff
@@ -163,6 +166,28 @@ func IsValidURL(str string, schemes []string, allowFragment bool) (bool, error)
 	return true, nil
 }
 
+// ContainsUrl determines if a string contains a URL anywhere inside it by breaking apart the string at certain
+// characters and inspecting each part.
+//
+// Uses IsValidURL and a subset of schemes to find a URL match.
+func ContainsUrl(fullString string) (bool, error) {
+	stringParts := strings.FieldsFunc(fullString, func(r rune) bool {
+		return r == ' ' || r == ',' || r == '#'
```

Comment:
This is not quite reliable as there's a large number of other characters (especially in Unicode) that can be used to separate strings that are not valid characters in a URL so that might bypass this method. E.g. I've confirmed that a payload like `https://example.com  👯` is "valid" while it should not be as it contains a URL (if you test this, make sure the spacer and emoji are allowed as valid characters in the validation pattern).
Would it be better to use a regex and find the first occurrence of a URL in the sting? That would probably be more reliable.

Reply:
Could be, but what we use as that regex? URL parsing is difficult, even on the easier ones. I'll do up a bunch of tests and see how using `BreakURLMatcher` compares, and then we can discuss further from there.

Reply:
If you'd still like to split the string, you have to split it by any character that is _not_ considered valid in a URL, probably not going to be very practical.
A regex like this comes up when Googling and might be worth a shot to see if you're tests are still passing: `https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)`

Reply:
I will happily give this a shot once I have more concrete ACs on the ticket as I am worried about what kinds of URLs we are looking for (example, would `ftp://` be valid? or what about more obscure schemes?) Once I have defined I will also give this a shot and be more than happy to try this and/or refine it 🙏